### ISSUE TITLE
Crash when returning delegate for libui functions

### ISSUE THREAD

**Issue by tom-corwin** [tom-corwin/tcdfx#15]
When running `SimpleWindowExample`, `SimpleWindowWithMenuExample` and I'm assuming `ControlGalleryExample`, the .NET Core Host (dotnet.exe) will crash, outputting:

```
C:\Program Files\dotnet\dotnet.exe (process 2932) exited with code -1073740940.
```

I debugged and after setting breakpoints at [NativeAssembly.cs#L54](https://github.com/tom-corwin/tcdfx/blob/52b4f567f47539c6da4bb646bfb164d2f28c4079/source/TCD.InteropServices/src/TCD/InteropServices/NativeAssembly.cs#L54), [NativeAssembly.cs#L55](https://github.com/tom-corwin/tcdfx/blob/52b4f567f47539c6da4bb646bfb164d2f28c4079/source/TCD.InteropServices/src/TCD/InteropServices/NativeAssembly.cs#L55), [NativeAssembly.cs#L57](https://github.com/tom-corwin/tcdfx/blob/52b4f567f47539c6da4bb646bfb164d2f28c4079/source/TCD.InteropServices/src/TCD/InteropServices/NativeAssembly.cs#L57), [Application.cs#L111](https://github.com/tom-corwin/tcdfx/blob/52b4f567f47539c6da4bb646bfb164d2f28c4079/source/TCD.UI/src/TCD/UI/Application.cs#L111), I found that .NET is crashing out when `NativeAssembly.LoadFunction<T>()` is returning at line [NativeAssembly.cs#L57](https://github.com/tom-corwin/tcdfx/blob/52b4f567f47539c6da4bb646bfb164d2f28c4079/source/TCD.InteropServices/src/TCD/InteropServices/NativeAssembly.cs#L57).

I debugged the crash dumps from `dotnet.exe` in Visual Studio (2019) using 'Mixed',  and got the following unhandled exception at [Application.cs#L111](https://github.com/tom-corwin/tcdfx/blob/52b4f567f47539c6da4bb646bfb164d2f28c4079/source/TCD.UI/src/TCD/UI/Application.cs#L111)  details:

```
Unhandled exception at 0x00007FF9C787AF49 (ntdll.dll) in dotnet.exe.192.dmp: 0xC0000374: A heap has been corrupted (parameters: 0x00007FF9C78E27F0).
```

I don't know if this affecting other platforms (Linux, macOS), but it's pretty frustrating, since the return value isn't null either.

**Comment by tom-corwin** [comment]
I need to investigate this more, since I have made an example project (to be uploaded) calling `Beep(uint, uint)` from `kernel32.dll` using `NativeAssembly`, and it works fine. It might just be corrupted  `libui` libraries or some typo that I'm missing.

**Comment by tom-corwin** [comment]
It's not that the library is corrupt. I replaced it with numerous other packages and still got the error. 

Also, this seems to only affect Windows platforms (8.1 and 10, that I can confirm).

**Comment by tom-corwin** [comment]
I'm starting to think that it's the `ntdll.dll` library on Windows that's causing the errors, but I can't say for sure.

**Comment by tom-corwin** [comment]
Well it seemed to be a matter of adding `[STAThread]` to all `Main()` methods in the examples that call `Application.Run()`. Fixed.

**Comment by tom-corwin** [comment]
Fixed with fdd9b02dd5a9690fbbcce14db1e87d39d5211786
